cpp/src/arrow/util/bit_stream_utils.h

```diff
@@ -423,6 +427,25 @@ inline bool BitReader::GetAligned(int num_bytes, T* v) {
   return true;
 }
 
+inline bool BitReader::Advance(int64_t num_bits) {
+  int64_t bits_required = bit_offset_ + num_bits;
+  int64_t bytes_required = BitUtil::BytesForBits(bits_required);
+  if (ARROW_PREDICT_FALSE(bytes_required > max_bytes_ - byte_offset_)) {
+    return false;
+  }
+  byte_offset_ += static_cast<int>(bits_required >> 3);
+  bit_offset_ = static_cast<int>(bits_required & 7);
+  // Reset buffered_values_
```

Comment:
A similar snippet already appears 3 times (see `GetValue_`, `GetAligned`, `GetBatch`). Perhaps factor it out as a private method?